# Repeated_Measures

# A Simple Repeated Measure Example: Fecal Fat

**Background: **A lack of digestive enzymes in the intestines can cause absorption problems, leading to excess fat in the feces. Pancreatic enzymes can be given to ameliorate the problem. Data was collected using multiple types of pills, including placebo, a normal tablet, an uncoated capsule, and a coated capsule. Each patient was given each type of pill and their fecal fat grams per day was measured. 

Therefore, we can think of this as a repeated measures study, since each patient has four measurements. 

The only predictor in this case is the pill type, and the outcome is the (continuous) variable of fecal fat in g/day. 

**Purpose: **To identify what the best pill type is for reducing fecal fat

**Methods: **First, I will follow the example of the chapter and try an ANOVA test

**Conclusions: **

# Inits

## Imports

## ANOVA test

Using a one-way ANOVA test would allow us to determine whether these groups are significantly different. The one-way ANOVA F-test has a null hypothesis that the mean outcome is the same in all groups, with an alternative hypothesis that the means differ in at least two of the populations. 

$H_{0}: \mu_{placebo} = \mu_{tablet} = \mu_{uncoated\ capsule} = \mu_{coated\ capsule}$

$H_{A}: \mu_{placebo} \neq \mu_{tablet} \neq \mu_{uncoated\ capsule} \neq \mu_{coated\ capsule}$

To use this ANOVA test, I will first load the data, then run a one-way ANOVA F-test on it

### Load the data

In [3]:
fec_fat_df = read.csv('./data/fecfat.csv')

In [5]:
head(fec_fat_df)

fecfat subject pilltype
1 44.5   1       none    
2  7.3   1       tablet  
3  3.4   1       capsule 
4 12.4   1       coated  
5 33.0   2       none    
6 23.1   2       capsule

### Run one-way ANOVA F-test

To run an ANOVA test, the group column is `pilltype`, and the response is the `fecfat`. Let's first check that the group column is formatted properly:

In [8]:
levels(fec_fat_df$pilltype)

[1] "capsule" "coated"  "none"    "tablet"

This looks good. Now, let's calculate the ANOVA and then show the F-test results

In [9]:
res.aov = aov(fecfat ~ pilltype, data = fec_fat_df)

In [10]:
summary(res.aov)

            Df Sum Sq Mean Sq F value Pr(>F)
pilltype     3   2009   669.5   1.862  0.169
Residuals   20   7193   359.7               

This shows that the F-value is 1.86, giving a p-value of 0.169, which is not significant